include/drivers/w1.h

```diff
@@ -0,0 +1,330 @@
+ *
+ * @param dev  Pointer to the device structure for the driver instance.
+ * @param bit  Transmitting bit value 1 or 0.
+ *
+ * @return a negative error code on error
+ */
+__syscall int w1_write_bit(const struct device *dev, const bool bit);
+
+static inline int z_impl_w1_write_bit(const struct device *dev, bool bit)
+{
+	const struct w1_driver_api *api = dev->api;
+
+	return api->write_bit(dev, bit);
+}
+
+/**
+ * @brief Read a single byte from the 1-Wire device.
+ *
+ * @param dev  Pointer to the device structure for the driver instance.
+ *
+ * @return The read byte value on success, or a negative error code on error.
+ */
+__syscall int w1_read_byte(const struct device *dev);
+
+static inline int z_impl_w1_read_byte(const struct device *dev)
+{
+	const struct w1_driver_api *api = dev->api;
+
+	return api->read_byte(dev);
+}
+
+/**
+ * @brief Write a single byte to the 1-Wire device.
+ *
+ * @param dev   Pointer to the device structure for the driver instance.
+ * @param byte  Transmitting byte.
+ *
+ * @return a negative error code on error
+ */
+__syscall int w1_write_byte(const struct device *dev, uint8_t byte);
+
+static inline int z_impl_w1_write_byte(const struct device *dev, uint8_t byte)
+{
+	const struct w1_driver_api *api = dev->api;
+
+	return api->write_byte(dev, byte);
+}
+
+/**
+ * @brief Read a block of data from the 1-Wire device.
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param buffer  Pointer to receive buffer.
+ * @param length  Length of receiving buffer (in bytes).
+ *
+ * @return a negative error code on error
+ */
+static inline int w1_read_block(const struct device *dev,
+				uint8_t *buffer, size_t length)
+{
+	int ret;
+
+	for (int i = 0; i < length; ++i) {
+		ret = w1_read_byte(dev);
+		if (ret < 0) {
+			return ret;
+		}
+		buffer[i] = ret;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Write a block of data from the 1-Wire device.
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param buffer  Pointer to transmitting buffer.
+ * @param length  Length of transmitting buffer (in bytes).
+ *
+ * @return a negative error code on error
+ */
+static inline int w1_write_block(const struct device *dev,
+				 const uint8_t *buffer, size_t length)
+{
+	int ret;
+
+	for (int i = 0; i < length; ++i) {
+		ret = w1_write_byte(dev, buffer[i]);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Get the number of peripheral devices on the bus.
+ *
+ * @param dev  Pointer to the device structure for the driver instance.
+ *
+ * @return Number of connected w1 peripheral device, or
+ *         a negative error code on error
+ */
+__syscall size_t w1_get_peripheral_count(const struct device *dev);
+
+static inline size_t z_impl_w1_get_peripheral_count(const struct device *dev)
+{
+	const struct w1_controller_config *ctrl_cfg = dev->config;
+
+	return ctrl_cfg->client_count;
+}
+
+/**
+ * @brief Configure parameters of W1-Master.
```

Comment:
Commit log says:

```
    Instead of master/slave terminology controller/peripheral terms are
    used to refer to bus participants.
```

But 'master' and 'slave' appear many times in this PR. What's going on?